### DISCUSSION TITLE
AWS/EC2 - shouldn't we just build bin/ena.usb?

### DISCUSSION THREAD

**Discussion by ech68** [ipxe/ipxe#803] [Q&A]
I know the docs mention building bin/ipxe.usb for use in AWS/EC2... but I've seen cases where (for some reason) a mysterious additional NIC is discovered by the iPXE AMI during boot (an eepro device that's got the broadcast address (all "ff"s) for its MAC) and iPXE then tries to initialize and DHCP from that device and hangs indefinitely, never falling through to the ENA device.

Given that all the instances I've seen in AWS use the ENA driver - wouldn't just building bin/ena.usb and using that as the AWS AMI work properly as well as avoid the failure I've described above, because the eepro support wouldn't be compiled into the AMI?

**Comment by mcb30** [comment]
There are also AWS instances types using the Xen paravirtual NIC (netfront) and some variants of Intel virtual functions (definitely intelxvf, possibly others). So ena.usb alone wouldn't be a general-purpose image for AWS.

**Reply by mcb30** [reply]
Options would be to either find the full list of AWS device types and build that, or to find the bug in the eepro driver.

**Comment by ech68** [comment]
I see - I guess the instances types we're currently using all have the ENA driver, but it's not necessarily everywhere.

As far as fixing the eepro driver - maybe if the mac address (regardless of driver) of a detected nic is "ff:ff:ff:ff:ff:ff" then it shouldn't be used as a potential DHCP target and should be skipped.

I've had no luck in reproducing the appearance of the eepro device on our instances, so it's difficult to test much.
